features/filesystem/storagelite/source/StorageLite.h

```diff
@@ -0,0 +1,559 @@
+     *          STORAGELITE_DATA_CORRUPT      Data is corrupt.
+     *          STORAGELITE_BAD_VALUE         Bad value in any of the parameters.
+     *          STORAGELITE_BUFF_TOO_SMALL    Not enough memory in user buffer.
+     */
+    int get_next_file(uint8_t *file_name, size_t file_name_buf_size,
+                      size_t& actual_file_name_size, uint32_t& handle);
+
+    /**
+     * @brief Initializes StorageLite component.
+     *
+     * @param[in]  bd                     Underlying block device.
+     * @param[in]  max_files              Maximal number of files.
+     *
+     * @returns STORAGELITE_SUCCESS       Initialization completed successfully.
+     *          STORAGELITE_READ_ERROR    Physical error reading data.
+     *          STORAGELITE_WRITE_ERROR   Physical error writing data (on recovery).
+     */
+    int init(BlockDevice *bd, size_t max_files = STORAGELITE_MAX_FILES);
+
+    /**
+     * @brief Deinitializes StorageLite component.
+     *        Warning: This function is not thread safe and should not be called
+     *        concurrently with other StorageLite functions.
+     *
+     * @returns STORAGELITE_SUCCESS       Deinitialization completed successfully.
+     */
+    int deinit();
+
+    /**
+     * @brief Reset StorageLite areas (delete all data).
+     *
+     * @returns STORAGELITE_SUCCESS       Reset completed successfully.
+     *          STORAGELITE_READ_ERROR    Physical error reading data.
+     *          STORAGELITE_WRITE_ERROR   Physical error writing data.
+     */
+    int reset();
+
+    /**
+     * @brief Reset StorageLite to factory defaults. i.e. copy all files defined as
+     *        backup copies on top of their active copies.
+     *
+     * @returns STORAGELITE_SUCCESS       Reset completed successfully.
+     *          STORAGELITE_READ_ERROR    Physical error reading data.
+     *          STORAGELITE_WRITE_ERROR   Physical error writing data.
+     */
+    int factory_reset();
+
+    /**
+     * @brief Return StorageLite size (area size).
+     *
+     * @returns StorageLite size.
+     */
+    size_t size() const;
+
+    /**
+     * @brief Return StorageLite free size.
+     *
+     * @returns StorageLite free size.
+     */
+    size_t free_size() const;
+
+    /**
+     * @brief Return address and size of a StorageLite area.
+     *
+     * @param[in]  area                   Area.
+     * @param[out] address                Area address.
+     * @param[out] size                   Area size (bytes).
+     *
+     * @returns STORAGELITE_SUCCESS           Success.
+     *          STORAGELITE_BAD_VALUE         Bad area parameter.
+     */
+    int get_area_params(uint8_t area, uint32_t& address, size_t& size);
+
+    static const uint32_t rollback_protect_flag;
+    static const uint32_t encrypt_flag;
+    static const uint32_t update_backup_flag;
+
+    static const size_t max_name_size;
+    static const size_t max_data_size;
```

Comment:
Are you referring to the flags? If so, then s we can use a combination of these, they should certainly not be defined as enums. As for them being all caps, I guess it would be true if they were defined with `#define`, but not sure whether it's the case if they are defined as `const` (maybe it's another C vs C++ argument). Do we have any rules for that? If so, will change these according to these rules.

Reply:
Fair point for enums.

In regards to name style: 
@0xc0170? My vote is all caps for any thing that is effectively a constant. This also makes it easier to switch between #defines and const variables if needed.